src/Physics/babylon.physicsEngine.ts

```diff
@@ -166,6 +166,128 @@
+                    strength,
+                    falloff
+                );
+                if (impostorForceAndContactPoint === false) {
+                    continue;
+                }
+
+                impostor.applyForce(
+                    impostorForceAndContactPoint.force,
+                    impostorForceAndContactPoint.contactPoint
+                );
+            }
+
+            return null;
+        }
+
+        private getImpostorForceAndContactPoint(impostor: PhysicsImpostor, origin: Vector3, radius: number, strength: number, falloff: PhysicsRadialImpulseFallof) {
+            if (impostor.mass === 0) {
+                return false;
+            }
+
+            if (!this.intersectsWithRadialSphere(impostor, origin, radius)) {
+                return false;
+            }
+
+            var impostorObject = (<Mesh>impostor.object);
+            var impostorObjectCenter = impostor.getObjectCenter();
+            var direction = impostorObjectCenter.subtract(origin);
+
+            var ray = new Ray(origin, direction, radius);
+            var hit = ray.intersectsMesh(impostorObject);
+
+            var contactPoint = hit.pickedPoint;
+            if (!contactPoint) {
+                return false;
+            }
+
+            var distanceFromOrigin = BABYLON.Vector3.Distance(origin, contactPoint);
+            if (distanceFromOrigin > radius) {
+                return false;
+            }
+            
+            var multiplier = falloff === PhysicsRadialImpulseFallof.Constant
+                ? strength
+                : strength * (1 - (distanceFromOrigin / radius));
+
+            var force = direction.multiplyByFloats(multiplier, multiplier, multiplier);
+
+            return { force: force, contactPoint: contactPoint };
+        }
+
+        private _radialSphere: Mesh; // create a sphere, so we can get the intersecting meshes inside
+        private _prepareRadialSphere(scene: Scene) {
+            if (!this._radialSphere) {
+                this._radialSphere = BABYLON.Mesh.CreateSphere(
+                    "radialSphere",
+                    32,
+                    1,
+                    scene
+                );
+                this._radialSphere.isVisible = false;
+            }
+        }
+
+        private intersectsWithRadialSphere(impostor: PhysicsImpostor, origin: Vector3, radius: number): boolean {
+            var impostorObject = <Mesh>impostor.object;
+
+            this._prepareRadialSphere(impostorObject.getScene());
+
+            this._radialSphere.position = origin;
+            this._radialSphere.scaling = new Vector3(radius * 2, radius * 2, radius * 2);
+            this._radialSphere._updateBoundingInfo();
+            this._radialSphere.computeWorldMatrix(true);
+  
+            return this._radialSphere.intersectsMesh(
+                impostorObject,
+                true
+            );
+        }
+        
+        public getRadialSphere() {
+            return this._radialSphere;

```

Comment:
Why is radialSphere a class member? This way there is no way of creating two of them at the same time. Or at least no way of using this function correctly.

Reply:
Oh, that was just a quick workaround for yesterday's review, because initially, I had showRadialSphere() & hideRadialSphere() methods, but I removed them because they kinda polluted the physics engine class, so I replaced them with only that one method. It's only really used in the example (`showRadialSphere()`) so you can better visualise the "explosion". I'm sure there are better ways, but it was already late and I wanted to fix it quickly, so that was the only solution I could think of.

Ok, great! Yeah, wasn't too sure where to put it, but if there would be more such functionality to come, then it makes sense to put it into a separate class.

Reply:
As DK said - visualization doesn't belong is this class. I see why you did that, we need to find a better solution. The problem is that we are releasing 3.1 pretty soon, and if I merge this on the physics engine level, it would break backwards compatibility if I take it out of there. Which we don't like at all :)

Want to try creating an external PhysicsHelper ? Move those functions there?  If you don't have time I will delay it a bit and do it in the next day or so right after merging. Since, again, I find it an amazing addition :smile:

Thanks!!!

Reply:
Oh yeah, absolutely! Will refactor the stuff today in the evening and ping you for a review. :)
Can even think of few more effects I can do there, like a magnetic force. Also looks quite nice - discovered it yesterday by accident.